# **1. Chargement et téléchargement des libraries**

Installation du package TCGAbiolinks : à faire une fois

In [1]:
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

BiocManager::install(c(
  "TCGAbiolinks",
  "SummarizedExperiment"
))

install.packages(c("dplyr", "ggplot2", "pheatmap"))

'getOption("repos")' replaces Bioconductor standard repositories, see
'help("repositories", package = "BiocManager")' for details.
Replacement repositories:
    CRAN: https://cran.r-project.org

Bioconductor version 3.21 (BiocManager 1.30.27), R 4.5.1 (2025-06-13)

Warning message:
“package(s) not installed when version(s) same as or greater than current; use
  `force = TRUE` to re-install: 'TCGAbiolinks' 'SummarizedExperiment'”
Old packages: 'BH', 'circlize', 'future.apply', 'httr2', 'isoband', 'knitr',
  'leidenbase', 'parallelly', 'RcppArmadillo', 'reformulas', 'Seurat',
  'SeuratObject', 'shiny', 'tidyr', 'TMB', 'xfun', 'yaml', 'yulab.utils', 'zoo'




Les packages binaires téléchargés sont dans
	/var/folders/k4/d0pwylgd1tl4mf2_hqnm6l400000gn/T//RtmpdOeeCA/downloaded_packages


chargement des libraries

In [2]:
library(TCGAbiolinks)
library(SummarizedExperiment)
library(dplyr)
library(ggplot2)
library(pheatmap)

Le chargement a nécessité le package : MatrixGenerics

Le chargement a nécessité le package : matrixStats


Attachement du package : ‘MatrixGenerics’


Les objets suivants sont masqués depuis ‘package:matrixStats’:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
  

# **2. Chargement du dataset TCGA thyroide**

### _**téléchargement des données TCGA**_
- projet THCA : cancer de la thyroide 
- seulement les données de quantification d'expression génique et de transcriptome profiling 

In [ ]:
query <- GDCquery(
  project = "TCGA-THCA",
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  workflow.type = "STAR - Counts"
)

GDCdownload(query)
se <- GDCprepare(query)

--------------------------------------

o GDCquery: Searching in GDC database

--------------------------------------

Genome of reference: hg38

--------------------------------------------

oo Accessing GDC. This might take a while...

--------------------------------------------

ooo Project: TCGA-THCA

--------------------

oo Filtering results

--------------------

ooo By data.type

ooo By workflow.type

----------------

oo Checking data

----------------

ooo Checking if there are duplicated cases

ooo Checking if there are results for the query

-------------------

o Preparing output

-------------------


Of the 572 files for download 572 already exist.

All samples have been already downloaded



In [5]:
se

class: RangedSummarizedExperiment 
dim: 60660 572 
metadata(1): data_release
assays(6): unstranded stranded_first ... fpkm_unstrand fpkm_uq_unstrand
rownames(60660): ENSG00000000003.15 ENSG00000000005.6 ...
  ENSG00000288674.1 ENSG00000288675.1
rowData names(10): source type ... hgnc_id havana_gene
colnames(572): TCGA-DJ-A2Q6-01A-11R-A18C-07
  TCGA-FK-A3SE-01A-11R-A22L-07 ... TCGA-DJ-A2PX-01A-11R-A18C-07
  TCGA-EL-A3ZS-01A-12R-A23N-07
colData names(241): barcode patient ... paper_TERT_AC paper_TERT_Q

### _**load des données cliniques**_

In [10]:
clinique <- GDCquery_clinic(
  project = "TCGA-THCA",
  type = "clinical"
)

head(clinique)
names(clinique)

project   submitter_id synchronous_malignancy ajcc_pathologic_stage
1 TCGA-THCA TCGA-EM-A2OZ No                     Stage I              
2 TCGA-THCA TCGA-EM-A22J No                     Stage II             
3 TCGA-THCA TCGA-DJ-A2PR No                     Stage I              
4 TCGA-THCA TCGA-ET-A40S No                     Stage II             
5 TCGA-THCA TCGA-ET-A39M No                     Stage I              
6 TCGA-THCA TCGA-ET-A40Q No                     Stage I              
  days_to_diagnosis laterality created_datetime last_known_disease_status
1 0                 Midline    NA               NA                       
2 0                 Left       NA               NA                       
3 0                 Bilateral  NA               NA                       
4 0                 Left       NA               NA                       
5 0                 Left       NA               NA                       
6 0                 Right      NA               NA                       
  tissue_or_organ_of_origin age_at_diagnosis ⋯
1 Thyroid gland             24195            ⋯
2 Thyroid gland             20539            ⋯
3 Thyroid gland              9914            ⋯
4 Thyroid gland             22733            ⋯
5 Thyroid gland             12441            ⋯
6 Thyroid gland             14136            ⋯
  treatments_radiation_treatment_anatomic_sites
1 NULL                                         
2 NULL;Primary Tumor Field                     
3 NULL                                         
4 NULL                                         
5 NULL;Primary Tumor Field                     
6 NULL;Primary Tumor Field                     
  treatments_radiation_treatment_outcome
1 NA                                    
2 NA;Complete Response                  
3 NA                                    
4 Not Reported;NA                       
5 NA;Complete Response                  
6 NA;Unknown                            
  treatments_radiation_treatment_dose_units
1 NA                                       
2 NA;mCi                                   
3 NA                                       
4 NA                                       
5 NA                                       
6 NA;mCi                                   
  treatments_radiation_number_of_fractions
1 NA                                      
2 NA                                      
3 NA                                      
4 NA                                      
5 NA;1                                    
6 NA;1                                    
  treatments_radiation_radiosensitizing_agent
1 NA                                         
2 NA                                         
3 NA                                         
4 NA                                         
5 NA                                         
6 NA                                         
  treatments_radiation_course_number
1 NA                                
2 NA                                
3 NA                                
4 NA                                
5 NA                                
6 NA                                
  treatments_radiation_clinical_trial_indicator
1 NA                                           
2 NA                                           
3 NA                                           
4 NA                                           
5 NA                                           
6 NA                                           
  treatments_radiation_route_of_administration
1 NULL                                        
2 NULL                                        
3 NULL                                        
4 NULL                                        
5 NULL                                        
6 NULL                                        
  treatments_radiation_initial_disease_status bcr_patient_barcode
1 NA                                          TCGA-EM-A2OZ       
2 NA                                          TCGA-EM-A

[1] "project"                                           
 [2] "submitter_id"                                      
 [3] "synchronous_malignancy"                            
 [4] "ajcc_pathologic_stage"                             
 [5] "days_to_diagnosis"                                 
 [6] "laterality"                                        
 [7] "created_datetime"                                  
 [8] "last_known_disease_status"                         
 [9] "tissue_or_organ_of_origin"                         
[10] "age_at_diagnosis"                                  
[11] "primary_diagnosis"                                 
[12] "updated_datetime"                                  
[13] "prior_malignancy"                                  
[14] "year_of_diagnosis"                                 
[15] "state"                                             
[16] "prior_treatment"                                   
[17] "diagnosis_is_primary_disease"                      
[18] "days_to_last_known_disease_status"                 
[19] "ajcc_staging_system_edition"                       
[20] "ajcc_pathologic_t"                                 
[21] "days_to_recurrence"                                
[22] "morphology"                                        
[23] "ajcc_pathologic_n"                                 
[24] "ajcc_pathologic_m"                                 
[25] "residual_disease"                                  
[26] "classification_of_tumor"                           
[27] "tumor_focality"                                    
[28] "diagnosis_id"                                      
[29] "icd_10_code"                                       
[30] "site_of_resection_or_biopsy"                       
[31] "tumor_grade"                                       
[32] "progression_or_recurrence"                         
[33] "tumor_of_origin"                                   
[34] "cigarettes_per_day"                                
[35] "alcohol_history"                                   
[36] "exposure_id"                                       
[37] "exposure_type"                                     
[38] "alcohol_intensity"                                 
[39] "race"                                              
[40] "gender"                                            
[41] "ethnicity"                                         
[42] "vital_status"                                      
[43] "age_at_index"                                      
[44] "days_to_birth"                                     
[45] "year_of_birth"                                     
[46] "demographic_id"                                    
[47] "age_is_obfuscated"                                 
[48] "year_of_death"                                     
[49] "country_of_residence_at_enrollment"                
[50] "days_to_death"                                     
[51] "days_to_last_follow_up"                            
[52] "follow_ups_disease_response"                       
[53] "treatments_pharmaceutical_treatment_intent_type"   
[54] "treatments_pharmaceutical_treatment_id"            
[55] "treatments_pharmaceutical_treatment_type"          
[56] "treatments_pharmaceutical_treatment_or_therapy"    
[57] "treatments_pharmaceutical_therapeutic_agents"      
[58] "treatments_pharmaceutical_timepoint_category"      
[59] "treatments_pharmaceutical_prescribed_dose_units"   
[60] "treatments_pharmaceutical_pretreatment"            
[61] "treatments_pharmaceutical_prescribed_dose"         
[62] "treatments_pharmaceutical_days_to_treatment_end"   
[63] "treatments_pharmaceutical_treatment_dose"          
[64] "treatments_pharmaceutical_days_to_treatment_start" 
[65] "treatments_pharmaceutical_treatment_anatomic_sites"
[66] "treatments_pharmaceutical_treatment_outcome"       
[67] "treatments_pharmaceutical_treatment_dose_units"    
[68] "treatments_pharmaceutical_number_of_fractions"     
[69] "treatments_pharmaceutical_radiosensitizing_agent"  

# **3. exploration du dataset**

In [11]:
# Dimensions
se
dim(se)

class: RangedSummarizedExperiment 
dim: 60660 572 
metadata(1): data_release
assays(6): unstranded stranded_first ... fpkm_unstrand fpkm_uq_unstrand
rownames(60660): ENSG00000000003.15 ENSG00000000005.6 ...
  ENSG00000288674.1 ENSG00000288675.1
rowData names(10): source type ... hgnc_id havana_gene
colnames(572): TCGA-DJ-A2Q6-01A-11R-A18C-07
  TCGA-FK-A3SE-01A-11R-A22L-07 ... TCGA-DJ-A2PX-01A-11R-A18C-07
  TCGA-EL-A3ZS-01A-12R-A23N-07
colData names(241): barcode patient ... paper_TERT_AC paper_TERT_Q

[1] 60660   572

In [12]:
# Métadonnées
meta <- as.data.frame(colData(se))
head(meta)

barcode                      patient     
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA
TCGA-FY-A2QD-01A-11R-A19O-07 TCGA-FY-A2QD-01A-11R-A19O-07 TCGA-FY-A2QD
TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3GR
TCGA-EL-A3T7-01A-11R-A22L-07 TCGA-EL-A3T7-01A-11R-A22L-07 TCGA-EL-A3T7
                             sample           shortLetterCode
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6-01A TP             
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE-01A TP             
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA-01A TP             
TCGA-FY-A2QD-01A-11R-A19O-07 TCGA-FY-A2QD-01A TP             
TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3GR-01A TP             
TCGA-EL-A3T7-01A-11R-A22L-07 TCGA-EL-A3T7-01A TP             
                             definition          sample_submitter_id
TCGA-DJ-A2Q6-01A-11R-A18C-07 Primary solid Tumor TCGA-DJ-A2Q6-01A   
TCGA-FK-A3SE-01A-11R-A22L-07 Primary solid Tumor TCGA-FK-A3SE-01A   
TCGA-DJ-A2QA-01A-11R-A18C-07 Primary solid Tumor TCGA-DJ-A2QA-01A   
TCGA-FY-A2QD-01A-11R-A19O-07 Primary solid Tumor TCGA-FY-A2QD-01A   
TCGA-EL-A3GR-01A-21R-A206-07 Primary solid Tumor TCGA-EL-A3GR-01A   
TCGA-EL-A3T7-01A-11R-A22L-07 Primary solid Tumor TCGA-EL-A3T7-01A   
                             tumor_descriptor
TCGA-DJ-A2Q6-01A-11R-A18C-07 Primary         
TCGA-FK-A3SE-01A-11R-A22L-07 Primary         
TCGA-DJ-A2QA-01A-11R-A18C-07 Primary         
TCGA-FY-A2QD-01A-11R-A19O-07 Primary         
TCGA-EL-A3GR-01A-21R-A206-07 Primary         
TCGA-EL-A3T7-01A-11R-A22L-07 Primary         
                             sample_id                           
TCGA-DJ-A2Q6-01A-11R-A18C-07 74b9c0c8-ef36-4194-ab8c-d81d61a39705
TCGA-FK-A3SE-01A-11R-A22L-07 d3cc158d-413d-49e6-8d23-3ed89fdea94a
TCGA-DJ-A2QA-01A-11R-A18C-07 2277f3f2-411e-47ab-a2b9-f80fd33540a1
TCGA-FY-A2QD-01A-11R-A19O-07 08d3cb3b-9a80-4034-b06d-b37dd38c8b9b
TCGA-EL-A3GR-01A-21R-A206-07 12693f53-fa09-494c-82eb-bf735ea84e98
TCGA-EL-A3T7-01A-11R-A22L-07 76e3d9f4-513f-4a44-8918-4c29ce5fb305
                             pathology_report_uuid                submitter_id
TCGA-DJ-A2Q6-01A-11R-A18C-07 CCF825BD-7410-4ED2-B0E2-8B17BB5C5BBD TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07 3489196F-F9FD-4797-B4F2-0D66D420102A TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07 E3CC8058-B202-4433-8597-6BD196D04AAC TCGA-DJ-A2QA
TCGA-FY-A2QD-01A-11R-A19O-07 D4447081-2054-46B6-B83F-C0D2DF8C4AA1 TCGA-FY-A2QD
TCGA-EL-A3GR-01A-21R-A206-07 F2945CC2-AEE4-45F7-A640-4174EB0D37FD TCGA-EL-A3GR
TCGA-EL-A3T7-01A-11R-A22L-07 6B1080F8-AF82-4FE3-B91D-3CA2DBA4A37E TCGA-EL-A3T7
                             ⋯ paper_mutDriver_DCCF_CI95H
TCGA-DJ-A2Q6-01A-11R-A18C-07 ⋯ 1,000                     
TCGA-FK-A3SE-01A-11R-A22L-07 ⋯                           
TCGA-DJ-A2QA-01A-11R-A18C-07 ⋯ 1,000                     
TCGA-FY-A2QD-01A-11R-A19O-07 ⋯                           
TCGA-EL-A3GR-01A-21R-A206-07 ⋯ 1,000                     
TCGA-EL-A3T7-01A-11R-A22L-07 ⋯ 0,982                     
                             paper_mutDriver_CCF_gene paper_EXCLUDE_SAMPLE
TCGA-DJ-A2Q6-01A-11R-A18C-07 BRAF:1.000               0                   
TCGA-FK-A3SE-01A-11R-A22L-07                          0                   
TCGA-DJ-A2QA-01A-11R-A18C-07 BRAF:1.000               0                   
TCGA-FY-A2QD-01A-11R-A19O-07                          0                   
TCGA-EL-A3GR-01A-21R-A206-07 BRAF:0.910               0                   
TCGA-EL-A3T7-01A-11R-A22L-07 BRAF:0.850               0                   
                             paper_patient paper_TERT_DEPTH paper_TERT_POWER
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6  81005            1               
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE  44535            1               
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA  75617            1               
TCGA-FY-A2QD-01A-11R-A19O-07 TCGA-FY-A2

In [13]:
# Types d’échantillons
table(meta$sample_type)
table(meta$shortLetterCode)


         Metastatic       Primary Tumor Solid Tissue Normal 
                  8                 505                  59 


 NT  TM  TP 
 59   8 505 

# **4. Création de la matrice d'expression**

## *Loading de la matrice d'expresison brute*

In [16]:
expr <- assay(se)   # counts
head(expr)
dim(expr)

TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-FK-A3SE-01A-11R-A22L-07
ENSG00000000003.15 6156                         7191                        
ENSG00000000005.6     7                            3                        
ENSG00000000419.13 2540                         3815                        
ENSG00000000457.14 1452                          767                        
ENSG00000000460.17  297                          276                        
ENSG00000000938.13 1075                         1069                        
                   TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-FY-A2QD-01A-11R-A19O-07
ENSG00000000003.15 2219                         5185                        
ENSG00000000005.6     1                            0                        
ENSG00000000419.13  735                         2594                        
ENSG00000000457.14  305                          970                        
ENSG00000000460.17   28                          135                        
ENSG00000000938.13  303                           71                        
                   TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3T7-01A-11R-A22L-07
ENSG00000000003.15 4466                         9702                        
ENSG00000000005.6     3                            0                        
ENSG00000000419.13 1496                         2506                        
ENSG00000000457.14  820                         1036                        
ENSG00000000460.17  208                          315                        
ENSG00000000938.13  610                         1576                        
                   TCGA-EL-A3T7-11A-21R-A22L-07 TCGA-FE-A230-01A-11R-A14Y-07
ENSG00000000003.15 4546                         2449                        
ENSG00000000005.6    10                            0                        
ENSG00000000419.13 1854                          811                        
ENSG00000000457.14 1279                          619                        
ENSG00000000460.17  183                          194                        
ENSG00000000938.13  459                          446                        
                   TCGA-EM-A3FO-01A-11R-A21D-07 TCGA-DJ-A3VB-01A-11R-A23N-07 ⋯
ENSG00000000003.15 6336                         5997                         ⋯
ENSG00000000005.6     0                            2                         ⋯
ENSG00000000419.13 1892                         3074                         ⋯
ENSG00000000457.14  725                          856                         ⋯
ENSG00000000460.17  149                          317                         ⋯
ENSG00000000938.13  315                         1829                         ⋯
                   TCGA-EM-A2P1-01A-11R-A206-07 TCGA-EL-A3T1-01A-11R-A22L-07
ENSG00000000003.15 2691                         4533                        
ENSG00000000005.6     0                            0                        
ENSG00000000419.13 2227                         2406                        
ENSG00000000457.14  759                         1120                        
ENSG00000000460.17  279                          381                        
ENSG00000000938.13 1147                         1127                        
                   TCGA-EL-A3T1-11A-11R-A22L-07 TCGA-DJ-A2PS-01A-11R-A18C-07
ENSG00000000003.15 2931                         4471                        
ENSG00000000005.6     5                            0                        
ENSG00000000419.13 2209                         1832                        
ENSG00000000457.14  607                          922                        
ENSG00000000460.17  169                          226                        
ENSG00000000938.13 1169                          440                        
                   TCGA-EL-A3TB-11A-11R-A22L-07 TCGA-EL-A3TB-01A-11R-A22L-07
ENSG00000000003.15 5029                         4742                        
ENSG00000000005.6     4                            3                        
E

[1] 60660   572

_commentaire_ : la matrice d'expression de base a 572 échantillons : 
- **8** "Metastatic"
- **59** "Solid Tissue Normal"
- **505** "Primary Tumor"

Mais les __*Solid Tissue Normal*__ ne sont pas des contrôles sains, juste des échantillons de tissus normaux prélevés à côté de la tumeur, ils n'ont pas de stade tumoral. Il faut qu'on les enlève pour ne garder que les échantillons "Primary Tumor".

Pour les __*Metastasique*__, ce ne sont pas des tissus thyroidiens, et ont un profil d'expression reprogrammé et souvent issus de stades avancés. Du coup le modèle va apprendre seulement le site métastasique plutôt que la biologie du stade tumoral. En fait, on fait ici du data leakage parce que les métastases impliquent forcément un stade avancé.

Alors, on ne garde que les __*Primary Tumor*__ pour faire l'analyse et pouvoir prédire les stades tumoraux. C'est ce que fait la cellule de code suivante :

In [ ]:
se_tumor <- se[, meta$sample_type == "Primary Tumor"] # ne garde que les tumeurs primaires
dim(se_tumor)

meta_tumor <- as.data.frame(colData(se_tumor)) # métadonnées des échantillons tumoraux
meta_tumor$patient_id <- substr(rownames(meta_tumor), 1, 12) # extraction des IDs patients

summary(table(meta_tumor$patient_id)) # permet de vérifier qu'il n'y a qu'un échantillon par patient, évite les doublons
# on peut maintenant dire que 1 colonne = 1 patient = 1 échantillon ssi "Number of factors = 1"

[1] 60660   505

Number of cases in table: 505 
Number of factors: 1 

## *Travailler sur un dataframe réduit, avec seulement les tumeurs primaires*

In [37]:
stopifnot(all(table(meta_tumor$patient_id) == 1)) # si il y a des doublons, le script s'arrête

se_patient <- se_tumor # simplifie les noms (pour pas s'y perdre)
metadata_patients <- meta_tumor

- se_patient : matrice d'expression avec seulement les échantillons "Primary Tumor"
- metadata_patient : metadata avec seulement les échantillons "Primary Tumor"

In [46]:
head(assay(se_patient))
dim(assay(se_patient))
head(metadata_patients, 3) # les 3 premiers

TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-FK-A3SE-01A-11R-A22L-07
ENSG00000000003.15 6156                         7191                        
ENSG00000000005.6     7                            3                        
ENSG00000000419.13 2540                         3815                        
ENSG00000000457.14 1452                          767                        
ENSG00000000460.17  297                          276                        
ENSG00000000938.13 1075                         1069                        
                   TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-FY-A2QD-01A-11R-A19O-07
ENSG00000000003.15 2219                         5185                        
ENSG00000000005.6     1                            0                        
ENSG00000000419.13  735                         2594                        
ENSG00000000457.14  305                          970                        
ENSG00000000460.17   28                          135                        
ENSG00000000938.13  303                           71                        
                   TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3T7-01A-11R-A22L-07
ENSG00000000003.15 4466                         9702                        
ENSG00000000005.6     3                            0                        
ENSG00000000419.13 1496                         2506                        
ENSG00000000457.14  820                         1036                        
ENSG00000000460.17  208                          315                        
ENSG00000000938.13  610                         1576                        
                   TCGA-FE-A230-01A-11R-A14Y-07 TCGA-EM-A3FO-01A-11R-A21D-07
ENSG00000000003.15 2449                         6336                        
ENSG00000000005.6     0                            0                        
ENSG00000000419.13  811                         1892                        
ENSG00000000457.14  619                          725                        
ENSG00000000460.17  194                          149                        
ENSG00000000938.13  446                          315                        
                   TCGA-DJ-A3VB-01A-11R-A23N-07 TCGA-BJ-A28X-01A-11R-A22L-07 ⋯
ENSG00000000003.15 5997                         3534                         ⋯
ENSG00000000005.6     2                            1                         ⋯
ENSG00000000419.13 3074                         1769                         ⋯
ENSG00000000457.14  856                          948                         ⋯
ENSG00000000460.17  317                          332                         ⋯
ENSG00000000938.13 1829                         1886                         ⋯
                   TCGA-DJ-A13S-01A-11R-A10U-07 TCGA-EM-A3SY-01A-11R-A22U-07
ENSG00000000003.15 2448                         3693                        
ENSG00000000005.6     0                            0                        
ENSG00000000419.13 1851                         2286                        
ENSG00000000457.14  764                          459                        
ENSG00000000460.17  176                          185                        
ENSG00000000938.13  635                          332                        
                   TCGA-EM-A2P1-01A-11R-A206-07 TCGA-EL-A3T1-01A-11R-A22L-07
ENSG00000000003.15 2691                         4533                        
ENSG00000000005.6     0                            0                        
ENSG00000000419.13 2227                         2406                        
ENSG00000000457.14  759                         1120                        
ENSG00000000460.17  279                          381                        
ENSG00000000938.13 1147                         1127                        
                   TCGA-DJ-A2PS-01A-11R-A18C-07 TCGA-EL-A3TB-01A-11R-A22L-07
ENSG00000000003.15 4471                         4742                        
ENSG00000000005.6     0                            3                        
E

[1] 60660   505

barcode                      patient     
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA
                             sample           shortLetterCode
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6-01A TP             
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE-01A TP             
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA-01A TP             
                             definition          sample_submitter_id
TCGA-DJ-A2Q6-01A-11R-A18C-07 Primary solid Tumor TCGA-DJ-A2Q6-01A   
TCGA-FK-A3SE-01A-11R-A22L-07 Primary solid Tumor TCGA-FK-A3SE-01A   
TCGA-DJ-A2QA-01A-11R-A18C-07 Primary solid Tumor TCGA-DJ-A2QA-01A   
                             tumor_descriptor
TCGA-DJ-A2Q6-01A-11R-A18C-07 Primary         
TCGA-FK-A3SE-01A-11R-A22L-07 Primary         
TCGA-DJ-A2QA-01A-11R-A18C-07 Primary         
                             sample_id                           
TCGA-DJ-A2Q6-01A-11R-A18C-07 74b9c0c8-ef36-4194-ab8c-d81d61a39705
TCGA-FK-A3SE-01A-11R-A22L-07 d3cc158d-413d-49e6-8d23-3ed89fdea94a
TCGA-DJ-A2QA-01A-11R-A18C-07 2277f3f2-411e-47ab-a2b9-f80fd33540a1
                             pathology_report_uuid                submitter_id
TCGA-DJ-A2Q6-01A-11R-A18C-07 CCF825BD-7410-4ED2-B0E2-8B17BB5C5BBD TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07 3489196F-F9FD-4797-B4F2-0D66D420102A TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07 E3CC8058-B202-4433-8597-6BD196D04AAC TCGA-DJ-A2QA
                             ⋯ paper_mutDriver_CCF_gene paper_EXCLUDE_SAMPLE
TCGA-DJ-A2Q6-01A-11R-A18C-07 ⋯ BRAF:1.000               0                   
TCGA-FK-A3SE-01A-11R-A22L-07 ⋯                          0                   
TCGA-DJ-A2QA-01A-11R-A18C-07 ⋯ BRAF:1.000               0                   
                             paper_patient paper_TERT_DEPTH paper_TERT_POWER
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6  81005            1               
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE  44535            1               
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA  75617            1               
                             paper_TERT paper_TERT_AF paper_TERT_AC
TCGA-DJ-A2Q6-01A-11R-A18C-07                          NA           
TCGA-FK-A3SE-01A-11R-A22L-07                          NA           
TCGA-DJ-A2QA-01A-11R-A18C-07                          NA           
                             paper_TERT_Q patient_id  
TCGA-DJ-A2Q6-01A-11R-A18C-07              TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07              TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07              TCGA-DJ-A2QA

Note d'Elodie : j'ai pas fini de trier la suite du code, j'ai juste trié ce qu'il y avait avant

In [ ]:
sample_barcodes <- colnames(se_TN)
patient_id <- substr(sample_barcodes, 1, 12)

meta <- as.data.frame(colData(se_TN)) %>%
  mutate(patient_id = patient_id)

# Join clinique (selon la colonne patient/submitter id présente chez toi)
# Souvent c'est "submitter_id" ou "case_submitter_id" selon la table.
names(clin)

# Exemple si la colonne s'appelle submitter_id :
meta2 <- meta %>%
  left_join(clin, by = c("patient_id" = "submitter_id"))

In [ ]:
meta_tumor <- meta2 %>%
  filter(sample_type == "Primary Tumor") %>%
  arrange(patient_id, sample_barcodes) %>%   # ordre stable
  distinct(patient_id, .keep_all = TRUE)

se_patient <- se_TN[, meta_tumor$barcode]   # selon le nom de la colonne barcode

In [44]:
table(clin_sub$stage, useNA = "ifany")


  Stage I  Stage II Stage III  Stage IV Stage IVA Stage IVC      <NA> 
      285        52       113         2        47         6         2 

In [47]:
clin <- GDCquery_clinic("TCGA-THCA", type = "clinical")

clin_sub <- clin_sub %>%
  mutate(
    stage_label = case_when(
      grepl("^Stage I$", stage) ~ "I",
      grepl("^Stage II$", stage) ~ "II",
      grepl("^Stage III$", stage) ~ "III",
      grepl("^Stage IV", stage) ~ "IV",
      TRUE ~ NA_character_
    )
  ) %>%
  filter(!is.na(stage_label))

meta_final <- meta_1 %>%
  left_join(clin_sub, by = c("patient_id" = "submitter_id"))

In [48]:
table(clin_sub$stage_label, useNA = "ifany")


  I  II III  IV 
285  52 113  55 

In [41]:
clin_sub <- clin_sub %>%
  mutate(
    stage_label = case_when(
      is.na(stage) ~ "control",
      grepl("Stage I", stage) ~ "I",
      grepl("Stage II", stage) ~ "II",
      grepl("Stage III", stage) ~ "III",
      grepl("Stage IV", stage) ~ "IV"
    )
  )

table(clin_sub$stage_label)


control       I 
      2     505 

In [38]:
expr_final <- expr[, meta_final$barcode]
head(t(expr_final))

ENSG00000000003.15 ENSG00000000005.6
TCGA-4C-A93U-01A-11R-A39I-07 2967               12               
TCGA-BJ-A0YZ-01A-11R-A10U-07 4676               11               
TCGA-BJ-A0Z0-01A-11R-A10U-07 3163                2               
TCGA-BJ-A0Z2-01A-11R-A10U-07 2786                1               
TCGA-BJ-A0Z3-01A-11R-A13Y-07 5626                0               
TCGA-BJ-A0Z5-01A-11R-A10U-07 2424                4               
                             ENSG00000000419.13 ENSG00000000457.14
TCGA-4C-A93U-01A-11R-A39I-07 1160                549              
TCGA-BJ-A0YZ-01A-11R-A10U-07 1184                878              
TCGA-BJ-A0Z0-01A-11R-A10U-07 1307               1271              
TCGA-BJ-A0Z2-01A-11R-A10U-07 1330                667              
TCGA-BJ-A0Z3-01A-11R-A13Y-07 2195               1309              
TCGA-BJ-A0Z5-01A-11R-A10U-07 1374                757              
                             ENSG00000000460.17 ENSG00000000938.13
TCGA-4C-A93U-01A-11R-A39I-07 213                 560              
TCGA-BJ-A0YZ-01A-11R-A10U-07 124                 310              
TCGA-BJ-A0Z0-01A-11R-A10U-07 100                 136              
TCGA-BJ-A0Z2-01A-11R-A10U-07 130                 137              
TCGA-BJ-A0Z3-01A-11R-A13Y-07 351                1214              
TCGA-BJ-A0Z5-01A-11R-A10U-07 255                 709              
                             ENSG00000000971.16 ENSG00000001036.14
TCGA-4C-A93U-01A-11R-A39I-07 23783              2651              
TCGA-BJ-A0YZ-01A-11R-A10U-07  1835              2812              
TCGA-BJ-A0Z0-01A-11R-A10U-07   412              2315              
TCGA-BJ-A0Z2-01A-11R-A10U-07   200              1961              
TCGA-BJ-A0Z3-01A-11R-A13Y-07 11717              3720              
TCGA-BJ-A0Z5-01A-11R-A10U-07   661              2146              
                             ENSG00000001084.13 ENSG00000001167.14 ⋯
TCGA-4C-A93U-01A-11R-A39I-07 1631               1914               ⋯
TCGA-BJ-A0YZ-01A-11R-A10U-07 3763               1737               ⋯
TCGA-BJ-A0Z0-01A-11R-A10U-07 1930               2330               ⋯
TCGA-BJ-A0Z2-01A-11R-A10U-07 1544               1984               ⋯
TCGA-BJ-A0Z3-01A-11R-A13Y-07 1981               2767               ⋯
TCGA-BJ-A0Z5-01A-11R-A10U-07 2081               1914               ⋯
                             ENSG00000288661.1 ENSG00000288662.1
TCGA-4C-A93U-01A-11R-A39I-07 0                 0                
TCGA-BJ-A0YZ-01A-11R-A10U-07 0                 0                
TCGA-BJ-A0Z0-01A-11R-A10U-07 0                 0                
TCGA-BJ-A0Z2-01A-11R-A10U-07 0                 0                
TCGA-BJ-A0Z3-01A-11R-A13Y-07 0                 1                
TCGA-BJ-A0Z5-01A-11R-A10U-07 0                 0                
                             ENSG00000288663.1 ENSG00000288665.1
TCGA-4C-A93U-01A-11R-A39I-07 23                0                
TCGA-BJ-A0YZ-01A-11R-A10U-07 11                0                
TCGA-BJ-A0Z0-01A-11R-A10U-07 12                0                
TCGA-BJ-A0Z2-01A-11R-A10U-07 35                0                
TCGA-BJ-A0Z3-01A-11R-A13Y-07 34                0                
TCGA-BJ-A0Z5-01A-11R-A10U-07 16                0                
                             ENSG00000288667.1 ENSG00000288669.1
TCGA-4C-A93U-01A-11R-A39I-07 0                 0                
TCGA-BJ-A0YZ-01A-11R-A10U-07 0                 0                
TCGA-BJ-A0Z0-01A-11R-A10U-07 0                 0                
TCGA-BJ-A0Z2-01A-11R-A10U-07 0                 0                
TCGA-BJ-A0Z3-01A-11R-A13Y-07 0                 0                
TCGA-BJ-A0Z5-01A-11R-A10U-07 0                 0                
                             ENSG00000288670.1 ENSG00000288671.1
TCGA-4C-A93U-01A-11R-A39I-07 358               0                
TCGA-BJ-A0YZ-01A-11R-A10U-07 394               0                
TCGA-BJ-A0Z0-01A-11R-A10U-07 443               0                
TCGA-BJ-A0Z2-01A-11R-A10U-07 388               0    

In [43]:
meta_final[, c("patient_id", "age", "sex", "stage_simple", "tumor_type")]

ERROR: Error in `[.data.frame`(meta_final, , c("patient_id", "age", "sex", "stage_simple", : undefined columns selected


In [ ]:

write.table(
  t(expr_final),
  file = "X_expression.tsv",
  sep = "\t",
  quote = FALSE
)

write.csv(
  meta_final[, c("patient_id", "age", "sex", "stage_simple")],
  file = "y_clinical.csv",
  row.names = FALSE
)